lib/iris/cube.py

```diff
@@ -2205,12 +2205,11 @@ def new_cell_measure_dims(cm_):
             first_slice = next(slice_gen)
         except StopIteration:
             first_slice = None
-        if self._numpy_array is not None:
-            cube_data = self._numpy_array
-        elif self._dask_array is not None:
+
+        if self.has_lazy_data():
             cube_data = self._dask_array
         else:
-            raise ValueError('This cube has no data, slicing is not supported')
+            cube_data = self._numpy_array
```

Comment:
I think if we end up at this point having a cube with no data we're deeper in the sticky than a `ValueError` will help with! The rest of the code in the class makes it clear that we can never _construct_ a cube with no data, so if an end user has done something that means their cube no longer has any data then IMO they can dig their own way out of the sticky stuff...